# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../hydrogen/flac-113.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- hydrogen-0.9.3/src/lib/FLACFile.cpp_orig	2007-03-14 13:07:58.000000000 +0100
+++ hydrogen-0.9.3/src/lib/FLACFile.cpp	2007-03-14 13:13:39.000000000 +0100
@@ -164,14 +164,13 @@
 	}
 
 	set_metadata_ignore_all();
-	set_filename( sFilename.c_str() );
 
-	State s=init();
-	if( s != FLAC__FILE_DECODER_OK ) {
+	FLAC__StreamDecoderInitStatus s=init(sFilename.c_str() );
+	if(s!=FLAC__STREAM_DECODER_INIT_STATUS_OK) {
 		errorLog( "[load] Error in init()" );
 	}
 
-	if ( process_until_end_of_file() == false ) {
+	if ( process_until_end_of_stream() == false ) {
 		errorLog( "[load] Error in process_until_end_of_file()" );
 	}
 }
